capstone/src/components/User/Adventurer/FindAdventure/Map/Map.jsx

```diff
@@ -29,20 +34,42 @@ export default function Map() {
       </VStack>
     );
   }
+
+  function onHover(restaurant) {
+    setActiveRestaurant(restaurant);
+  }
+  function onLeave() {
+    setActiveRestaurant(null);
+  }
   return (
     <GoogleMap
       center={currentPosition}
-      zoom={12}
+      zoom={14}
       mapContainerClassName="mapContainer"
       options={{
         mapTypeControl: false,
         streetViewControl: false,
         maxZoom: 20,
-        minZoom: 15,
+        minZoom: 12,
       }}
     >
-      {restaurants.map((restaurant, index) => (
-        <MarkerF key={index} position={restaurant.geometry.location} />
+      {restaurants.map((restaurant) => (
+        <Fragment key={Math.random()}>
```

Comment:
Why are we doing this? This forces this component tree to refresh on every render. Maybe you can try passing in `restaurant.place_id` instead.